src/shuffle.c

```diff
@@ -92,6 +99,8 @@ int shuffle_merge(int num) {
         fid[fidcounter] = fopen(filename, "rb");
         if (fid[fidcounter] == NULL) {
             fprintf(stderr, "Unable to open file %s.\n",filename);
+            free(array);
+            free_fid(fid, num);
```

Comment:
Same issue as in the other review.  I believe that fid can have non-zero entries at this point, and the subsequent fclose call will be broken.

Reply:
Maybe just pass in fidcounter fixes that?

Reply:
I do feel strongly about this one though - I think this is broken as is.

Reply:
Updated.